### ISSUE TITLE
array-bracket-spacing conflicts with comma-spacing

### ISSUE THREAD

**Issue by kewisch** [eslint/eslint#6547]
**What version of ESLint are you using?** 2.13.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
"array-bracket-spacing": [2, "never"], 
"comma-spacing": [2, {"before": false, "after": true}],
"no-sparse-arrays": 2,
```

**What did you do? Please include the actual source code causing the issue.**

```
for (let [propName, ] of this.mProperties) {
 ...
}
```

**What did you expect to happen?**
`comma-spacing` asks for a space after the sparse array comma, `array-bracket-spacing` prevents this. In terms of sparse arrays I don't really have a preference, but if I had to choose, then without a space.

**What actually happened? Please include the actual, raw output from ESLint.**
`There should be no space before ']'  array-bracket-spacing`
-or-
`A space is required after ','  comma-spacing`

**Comment by platinumazure** [comment]
I bet we could add an option to `array-bracket-spacing` to support this (that is, to avoid enforcing when the first or last token is a comma and effectively defer to comma-spacing. That seems safer to me since comma-spacing has effects in objects and many other contexts, whereas this problem is unique to (sparse) arrays.

@eslint/eslint-team Thoughts?

**Comment by michaelficarra** [comment]
That is not a sparse array. It is just a trailing comma.

**Comment by kewisch** [comment]
I am not sure about the terminology, but if that is the case, maybe this example would be more meaningful (which is why I believed it to be a sparse array):
```javascript
let foo = [[1,2,3], [4,5,6]];
for (let [x, , z] of foo) {
  alert(x + "-" + z);
}
```

**Comment by nzakas** [comment]
We probably just need a separate `comma-spacing` option for trailing commas so you can alter the behavior in just that case.

**Comment by kewisch** [comment]
Sounds good to me. My mind finally managed to see it as a trailing comma now too, sorry for the noise. What would you suggest for naming?

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
